syntax = "proto3";
package arktouros.proto.collector.v1;

option java_multiple_files = true;
option java_package = "edu.npu.arktouros.proto.collector.v1";
option java_outer_classname = "MetricServiceProto";

import "model/metric.proto";

service MetricService {
  rpc Export(MetricRequest) returns (MetricResponse);
}

message MetricRequest {
  repeated Metric metrics = 1;
}

message Metric {
  oneof metric {
    arktouros.proto.metric.v1.Gauge gauge = 1;
    arktouros.proto.metric.v1.Counter counter = 2;
    arktouros.proto.metric.v1.Histogram histogram = 3;
    arktouros.proto.metric.v1.Summary summary = 4;
  }
}
message MetricResponse {
  // The number of rejected log records.
  //
  // A `rejected_<signal>` field holding a `0` value indicates that the
  // request was fully accepted.
  int64 rejected_metric_records = 1;

  // A developer-facing human-readable message in English. It should be used
  // either to explain why the server rejected parts of the data during a partial
  // success or to convey warnings/suggestions during a full success. The message
  // should offer guidance on how users can address such issues.
  //
  // error_message is an optional field. An error_message with an empty value
  // is equivalent to it not being set.
  string error_message = 2;
}